src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -114,57 +108,36 @@ const getMessageFromBraze = async (
 					undefined,
 					internalButtonId,
 				);
-				appboy.logInAppMessageButtonClick(thisButton, message);
+				appboy.logInAppMessageButtonClick(thisButton, message as any);
 			};
 
 			const logImpressionWithBraze = () => {
 				// Log the impression with Braze
 				appboy.logInAppMessageImpression(message);
 			};
-
-			if (extras) {
-				const meta = {
-					dataFromBraze: extras,
-					logImpressionWithBraze,
-					logButtonClickWithBraze,
-				};
-
-				resolve({ result: true, meta });
-			} else {
-				resolve({ result: false });
-			}
-
-			// Unsubscribe
-			if (subscriptionId) {
-				appboy.removeSubscription(subscriptionId);
-			}
-		};
-
-		// Keep hold of the subscription ID so that we can unsubscribe in the
-		// callback, ensuring that the callback is only invoked once per page
-		subscriptionId = appboy.subscribeToInAppMessage(callback);
-
-		setHasCurrentBrazeUser();
-		appboy.changeUser(brazeUuid);
-		appboy.openSession();
-	});
-
-	canShowPromise
-		.then(() => {
+			const meta = {
+				dataFromBraze: message.extras,
+				logImpressionWithBraze,
+				logButtonClickWithBraze,
+			};
+			return { result: true, meta };
+		})
+		.then((outcome) => {
 			const appboyTimeTaken = appboyTiming.end();
 
 			record({
 				component: 'braze-appboy-timing',
 				value: appboyTimeTaken,
 			});
+
+			return outcome;
 		})
 		.catch(() => {
 			appboyTiming.clear();
 			// eslint-disable-next-line no-console
 			console.log('Appboy Timing failed.');
+			return { result: false };
```

Comment:
I think returning `{ result: false }` here is the right thing to do. It occurs to me that the `console.log` just above isn't entirely correct is it? The promise chain could have rejected for any reason, not just that the timing failed, isn't that right?

Reply:
Yes true. I'm not sure there's anything particularly useful we can log here given that - so I've removed it.